src/vg/civcraft/mc/civmenu/TOSListener.java

```diff
@@ -66,13 +65,27 @@ public void run() {
 	public void playerMoveEvent(PlayerMoveEvent event) {
 		Player p = event.getPlayer();
 		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
-			if(event.getTo().distance(locations.get(p.getUniqueId())) > plugin.GetConfig().get("terms.MovementRange").getInt()){
+			Location from = locations.get(p.getUniqueId());
+			if (from == null){
+				from = event.getFrom();
+				locations.put(p.getUniqueId(), from);
+			}
+			if(event.getTo().distance(from) > plugin.GetConfig().get("terms.MovementRange").getInt()){
 				p.sendMessage(ChatColor.RED + "You must accept the terms in order to play.");
 				sendTOS(p);
-				event.setTo(locations.get(p.getUniqueId()));
+				event.setTo(from);
 			}
 		}
 	}
+	
+	@EventHandler
+	public void playerRespawnEvent(PlayerRespawnEvent e) {
+		Player p = e.getPlayer();
+		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
+			locations.remove(p.getUniqueId());
```

Comment:
No. It's not the spawning itself that puts me inside blocks, its the teleport from the move event that does it. I had this right after line 85:
```java
locations.put(p.getUniqueId(), e.getRespawnLocation());
```
With that added, the teleport when going too far puts you inside the ground. I am not sure why...